api/python/t4/snapshots.py

```diff
@@ -160,6 +160,46 @@ def hash_file(readable_file):
 
     return hasher.hexdigest()
 
+def bytes_to_int(input_bytes):
+    result = 0
+    for byte in input_bytes:
+        result = result * 256 + int(byte)
+    return result
+
+HASH_MAX = 2 ** 256
+
+def sum_hash(current, add_bytes):
+    """
+    Sums hashes for order-independence.
+
+    Args:
+        current(int)
+        add_bytes(bytes)
+
+    Returns:
+        New total
+    """
+    total = current + bytes_to_int(add_bytes)
+    total = total % HASH_MAX
+    return total
+
+def hash_object(obj):
```

Comment:
`PackageEntry.hash()`?

Reply:
The top hash of a Package intentionally excludes the physical keys, which is not necessarily a property of the PackageEntry. Also, PackageEntry objects are mutable (which I don't have a convenient way of preventing) so they don't have a stable hash.